src/osd/PrimaryLogPG.cc

```diff
@@ -10572,7 +10581,24 @@ void PrimaryLogPG::mark_all_unfound_lost(
 	  release_backoffs(p.first);
 	}
 	requeue_object_waiters(waiting_for_unreadable_object);
-	queue_recovery();
+	if (is_recovery_unfound()) {
+	  queue_peering_event(
+	    CephPeeringEvtRef(
+	      std::make_shared<CephPeeringEvt>(
+	      get_osdmap()->get_epoch(),
+	      get_osdmap()->get_epoch(),
+	      DoRecovery())));
+	} else if (is_backfill_unfound()) {
+	  queue_peering_event(
+	    CephPeeringEvtRef(
+	      std::make_shared<CephPeeringEvt>(
+	      get_osdmap()->get_epoch(),
+	      get_osdmap()->get_epoch(),
+	      RequestBackfill())));
+	} else {
+	  // Not sure if this is needed anymore
```

Comment:
I guess we could replace it with an assert to catch bugs where we're not in {backfill,recovery}_unfound